From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Wed, 11 Sep 2024 02:26:23 +0200
Subject: gdm/webLogin: Improve label rendering

---
 js/gdm/webLogin.js | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/js/gdm/webLogin.js b/js/gdm/webLogin.js
index 55e437f..244cab0 100644
--- a/js/gdm/webLogin.js
+++ b/js/gdm/webLogin.js
@@ -5,6 +5,7 @@
 
 import Clutter from 'gi://Clutter';
 import GObject from 'gi://GObject';
+import Pango from 'gi://Pango';
 import St from 'gi://St';
 import {QrCode} from '../ui/qrCode.js';
 import {Spinner} from '../ui/animation.js';
@@ -33,6 +34,10 @@ class WebLoginPrompt extends St.BoxLayout {
             text: message,
             style_class: 'web-login-title-label',
         });
+        this._urlTitleLabel.clutterText.set({
+            lineWrap: true,
+            ellipsize: Pango.EllipsizeMode.NONE,
+        });
         this.add_child(this._urlTitleLabel);
 
         const qrCode = new QrCode({
@@ -75,6 +80,9 @@ class WebLoginPrompt extends St.BoxLayout {
         const http = 'http://';
         const https = 'https://';
 
+        if (url.endsWith('/'))
+            url = url.substring(0, url.length-1);
+
         if (url.startsWith(http))
             return url.substring(http.length);
 
