From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 11 Jul 2024 03:01:26 +0200
Subject: gdm/authPrompt: Grab key focus when choice list is shown

Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-shell/+bug/2072716
---
 js/gdm/authPrompt.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/js/gdm/authPrompt.js b/js/gdm/authPrompt.js
index 88b596d..878917a 100644
--- a/js/gdm/authPrompt.js
+++ b/js/gdm/authPrompt.js
@@ -701,6 +701,7 @@ export const AuthPrompt = GObject.registerClass({
     setChoiceList(promptMessage, choiceList) {
         this._authList.clear();
         this._authList.label.text = promptMessage;
+        this._authList.grab_key_focus();
         for (let key in choiceList) {
             let text = choiceList[key];
             this._authList.addItem(key, text);
