From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 14 Aug 2025 19:59:13 +0200
Subject: js/environment: Add implementation for GDesktopAppIcon get_string

---
 js/ui/environment.js | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/js/ui/environment.js b/js/ui/environment.js
index a008a37..b3a03b7 100644
--- a/js/ui/environment.js
+++ b/js/ui/environment.js
@@ -351,6 +351,10 @@ if (!GioUnix.DesktopAppInfo.prototype.has_key) {
         return GioUnix.DesktopAppInfo.get_boolean(this, key);
     };
 
+    GioUnix.DesktopAppInfo.prototype.get_string = function (key) {
+        return GioUnix.DesktopAppInfo.get_string(this, key);
+    };
+
     GioUnix.DesktopAppInfo.prototype.get_categories = function () {
         return GioUnix.DesktopAppInfo.get_categories(this);
     };
